roles/sap_general_preconfigure/tasks/RedHat/generic/assert-etc-hosts.yml

```diff
@@ -20,17 +20,18 @@
     success_msg: "PASS: The line '{{ sap_general_preconfigure_ip }} {{ sap_general_preconfigure_hostname }}.{{ sap_general_preconfigure_domain }} {{ sap_general_preconfigure_hostname }}' is once in /etc/hosts."
   ignore_errors: "{{ sap_general_preconfigure_assert_ignore_errors | d(false) }}"
 
+# We allow more than one line containing sap_general_preconfigure_ip:
 - name: Count the number of sap_general_preconfigure_ip ({{ sap_general_preconfigure_ip }}) entries in /etc/hosts
-  ansible.builtin.command: awk 'BEGIN{a=0}/{{ sap_general_preconfigure_ip }}/{a++}END{print a}' /etc/hosts
+  ansible.builtin.command: awk 'BEGIN{a=0}/^{{ sap_general_preconfigure_ip }}\s/{a++}END{print a}' /etc/hosts
   register: __sap_general_preconfigure_register_sap_ip_once_assert
   ignore_errors: yes
   changed_when: no
 
-- name: Assert that there is just one line containing {{ sap_general_preconfigure_ip }} in /etc/hosts
+- name: Assert that there is at least one line containing {{ sap_general_preconfigure_ip }} in /etc/hosts
```

Comment:
As a sysadmin, I would not allow PASS-ing this check if one IP is configured more than once in /etc/hosts.

Reply:
We currently allow more than one line with the same IP address in /etc/hosts in https://github.com/sap-linuxlab/community.sap_install/blob/42af6baab0310ec9c47994147d877f3d11ca4108/roles/sap_maintain_etc_hosts/tasks/update_host_present.yml#L85, so we now should discuss the impact of allowing such a configuration in a broader context.

Reply:
Hi, 
IIRC this was a user request. This user is putting host aliases to separate lines instead of appending to a single line. This was in the very beginning and I am not sure if this has been proper documented. 
So I am fine with approving this PR and wait for complaints. Especially because I haven't found any evidence, that this is a common practise in the net.